packages/components/src/text-highlight/index.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { escapeRegExp } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createInterpolateElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { TextHighlightProps } from './types';
+
+/**
+ * Highlights occurrences of a given string within another string of text. Wraps
+ * each match with a `<mark>` tag which provides browser default styling.
+ *
+ * ```jsx
+ * import { TextHighlight } from '@wordpress/components';
+ *
+ * const MyTextHighlight = () => (
+ *   <TextHighlight
+ *     text="Why do we like Gutenberg? Because Gutenberg is the best!"
+ *     highlight="Gutenberg"
+ *   />
+ * );
+ * ```
+ */
+export const TextHighlight = ( props: TextHighlightProps ) => {
```

Comment:
This is an interesting example of how ref forwarding and using the `WordPressComponentProps` type don't make much sense with the current way the component is written (cc @mirka )